spec/solver_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'rspec'
+require_relative '../solver'
+
+describe Solver do
+  solver = Solver.new
+
+  context '#factorial' do
+    it 'returns the factorial of the number' do
+      expect(solver.factorial(5)).to eql(120)
+    end
+  end
```

Comment:
- [ ] As per this [requirement](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/testing/tdd_project.md#:~:text=to%20N%20and-,has%20the%20special%20case%20that%20the%20factorial%20of%200%20is%201.%20This%20method%20only%20accepts%200%20and%20positive%20integers%2C%20so%20if%20a%20negative%20integer%20is%20given%20it%20should%20raise%20an%20exception.,-Create%20a%20method), Please make sure that you cover all edge cases inside your tests: 
- if a negative integer is given it should raise an exception.
- special case that the factorial of 0 is 1.